hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/SyntheticTxnFactory.java

```diff
@@ -501,6 +504,42 @@ public TransactionBody.Builder createUnFreeze(
         return TransactionBody.newBuilder().setTokenUnfreeze(builder);
     }
 
+    public TransactionBody.Builder createTokenUpdate(TokenUpdateWrapper updateWrapper) {
+        final var builder = TokenUpdateTransactionBody.newBuilder();
+        builder.setToken(updateWrapper.tokenID())
+                .setName(updateWrapper.name())
+                .setSymbol(updateWrapper.symbol())
+                .setMemo(StringValue.of(updateWrapper.memo()));
+        if (updateWrapper.treasury() != null) builder.setTreasury(updateWrapper.treasury());
+
+        if (updateWrapper.expiry().second() != 0) {
+            builder.setExpiry(
+                    Timestamp.newBuilder().setSeconds(updateWrapper.expiry().second()).build());
+        }
+        if (updateWrapper.expiry().autoRenewAccount() != null)
+            builder.setAutoRenewAccount(updateWrapper.expiry().autoRenewAccount());
+        if (updateWrapper.expiry().autoRenewPeriod() != 0) {
+            builder.setAutoRenewPeriod(
+                    Duration.newBuilder().setSeconds(updateWrapper.expiry().autoRenewPeriod()));
+        }
+        updateWrapper
+                .tokenKeys()
+                .forEach(
+                        tokenKeyWrapper -> {
+                            final var key = tokenKeyWrapper.key().asGrpc();
+                            if (tokenKeyWrapper.isUsedForAdminKey()) builder.setAdminKey(key);
+                            if (tokenKeyWrapper.isUsedForKycKey()) builder.setKycKey(key);
+                            if (tokenKeyWrapper.isUsedForFreezeKey()) builder.setFreezeKey(key);
+                            if (tokenKeyWrapper.isUsedForWipeKey()) builder.setWipeKey(key);
+                            if (tokenKeyWrapper.isUsedForSupplyKey()) builder.setSupplyKey(key);
+                            if (tokenKeyWrapper.isUsedForFeeScheduleKey())
+                                builder.setFeeScheduleKey(key);
+                            if (tokenKeyWrapper.isUsedForPauseKey()) builder.setPauseKey(key);
```

Comment:
Probably this section can be moved to a function and re-used between `tokenCreate` and `tokenUpdate`

Reply:
I'm not sure if this is doable since the builders are different for both cases(TokenUpdateTransactionBody.Builder and 
TokenCreateTransactionBody.Builder) and don't share a common parent which has the key setters. If you have a proposal here please share it :).

Reply:
Yes, that makes sense.